tests/lib/rules/padding-line-between-statements.js

```diff
@@ -0,0 +1,4420 @@
+                { blankline: "never", prev: ["const", "let", "var"], next: "*" },
+                { blankline: "any", prev: ["const", "let", "var"], next: ["const", "let", "var"] }
+            ]]
+        },
+        {
+            code: "switch(a) {\ncase 0:\nvar foo;\ncase 1:}",
+            options: [[
+                { blankline: "always", prev: ["const", "let", "var"], next: "*" },
+                { blankline: "any", prev: ["const", "let", "var"], next: ["const", "let", "var"] }
+            ]]
+        },
+        {
+            code: `
+                var a = 1
+
+                ;
+                (b || c).doSomething();
+            `,
+            options: [[
+                { blankline: "never", prev: ["const", "let", "var"], next: "*" },
+                { blankline: "any", prev: ["const", "let", "var"], next: ["const", "let", "var"] }
+            ]]
+        },
+
+        //----------------------------------------------------------------------
+        // Tests from newline-before-return
+        //----------------------------------------------------------------------
+
+        {
+            code: "function a() {\nreturn;\n}",
+            options: [[
+                { blankline: "always", prev: "*", next: "return" }
+            ]]
+        },
+        {
+            code: "function a() {\n\nreturn;\n}",
+            options: [[
+                { blankline: "always", prev: "*", next: "return" }
+            ]]
+        },
+        {
+            code: "function a() {\nvar b;\n\nreturn;\n}",
+            options: [[
+                { blankline: "always", prev: "*", next: "return" }
+            ]]
+        },
+        {
+            code: "function a() {\nif (b) return;\n}",
+            options: [[
+                { blankline: "always", prev: "*", next: "return" }
+            ]]
+        },
+        {
+            code: "function a() {\nif (b) { return; }\n}",
+            options: [[
+                { blankline: "always", prev: "*", next: "return" }
+            ]]
+        },
+        {
+            code: "function a() {\nif (b) {\nreturn;\n}\n}",
+            options: [[
+                { blankline: "always", prev: "*", next: "return" }
+            ]]
+        },
+        {
+            code: "function a() {\nif (b) {\n\nreturn;\n}\n}",
+            options: [[
+                { blankline: "always", prev: "*", next: "return" }
+            ]]
+        },
+        {
+            code: "function a() {\nif (b) {\nreturn;\n}\n\nreturn c;\n}",
+            options: [[
+                { blankline: "always", prev: "*", next: "return" }
+            ]]
+        },
+        {
+            code: "function a() {\nif (b) {\n\nreturn;\n}\n\nreturn c;\n}",
+            options: [[
+                { blankline: "always", prev: "*", next: "return" }
+            ]]
+        },
+        {
+            code: "function a() {\nif (!b) {\nreturn;\n} else {\nreturn b;\n}\n}",
+            options: [[
+                { 
```

Comment:
I think it would be better to use constants for the error messages to avoid duplication.

```js
const UNEXPECTED_BLANK = "Unexpected blank lines before this statement.";

// ...

errors: [UNEXPECTED_BLANK]
```